protocol/persistence_legacy.go

```diff
@@ -313,6 +325,51 @@ func (db sqlitePersistence) MessagesExist(ids []string) (map[string]bool, error)
 	return result, nil
 }
 
+func (db sqlitePersistence) MessagesByIDs(ids []string) ([]*Message, error) {
+	if len(ids) == 0 {
+		return nil, nil
+	}
+
+	idsArgs := make([]interface{}, 0, len(ids))
```

Comment:
I don't think this is necessary. Can't we just pass `ids` to `Query()`?

Reply:
just checked, actually not sure we can, as `Query` expects `[]interface{}` rather than `[]string`,
I have made the conversion a bit faster (not using append,  and using index instead, although not sure it makes a difference here as the array is allocated already), is there any other way to do the conversion?

Reply:
You are right, sorry for the confusion. And no, there is no other way to case it :( It's the best way.